src/test/java/test/TestMoneroWalletGeneric.java

```diff
@@ -0,0 +1,26 @@
+package test;
+
+import org.junit.Test;
+
+import monero.daemon.model.MoneroNetworkType;
+import monero.utils.MoneroUtils;
+
+public class TestMoneroWalletGeneric {
+
+  @Test
+  public void testAddressValidation() {
```

Comment:
Tests must cover STAGENET.  Network type should either be specified or default to MAINNET.

Reply:
These tests should show that invalid addresses, including mixing up network type, will throw an exception.